src/twisted/newsfragments/7798.bugfix

```diff
@@ -0,0 +1 @@
+Initiating a TCP connection with an address literals and a port outside of the 1-65535 range in twisted.internet.tcp.BaseClient and twisted.internet.iocpreactor.tcp.Client now triggers an errback instead of an unhandled error. This change might also fix other unhandled errors occurred while connecting using address literals.
```

Comment:
Thinking of `BaseClient` and `Client` as public API is correct but it still makes me grind my teeth a bit 😬 . Could we describe this as `IReactorTCP.connectTCP` implementations.

Reply:
Sure.

I have updated to comply with our newsfragment standard.

This code was originally written a long time ago.... 2018 

```suggestion
IReactorTCP.connectTCP via the twisted.internet.tcp.BaseClient and twisted.internet.iocpreactor.tcp.Client implementations,
now triggers an errback instead of an unhandled error when using a TCP connection with an address literals and a port outside of the 1-65535 range. This change might also fix other unhandled errors occurred while connecting using address literals.
```